### ISSUE TITLE
Enhance functional testing to verify report links are valid

### ISSUE THREAD

**Issue by crutchfield** [cisagov/ScubaGear#850]
# 💡 Summary #

The ScubaGear reports contain links to external documents.  Currently the automated tests do not check that they are valid.  We should enhance the functional tests to include this check. 

## Motivation and context ##

This would be useful because we have encounter bugs multiple times where the links were incorrect.  Adding the check will allow us to detect issues earlier in the future.

## Implementation notes ##

Please provide details for implementation, such as:

- validate header links (SCUBA, Baseline documents) in main document
- validate policy group links in individual products (e.g., TEAMS-1 Meeting Policies)
- validate embedded links (e.g., See Secure Configuration Baseline policy) for instructions on manual check.

## Acceptance criteria ##

How do we know when this work is done?

- [ ] Functional testing include checks for links and pass

**Comment by mitchelbaker-cisa** [comment]
Reference test_links.py file on ScubaGoggles side for inspiration, https://github.com/cisagov/ScubaGoggles/blob/352-add-markdown-link-validation-workflows/Testing/Markdown/test_links.py.

**Comment by FollyBeachGurl** [comment]
During refinement the team decided to check and see if this issue already exists.

**Comment by rksharma-owg** [comment]
I looked through the current test layout and this seems pretty implementable without inventing a new test harness.

A few repo-specific pointers that may help whoever picks it up:

- There is already a markdown-focused functional test suite at `Testing/Functional/MarkdownChecks/ValidateMarkdowns.Tests.ps1`.
- The repo already has a helper around `Invoke-WebRequest` retry behavior in `Testing/Functional/Products/FunctionalTestUtils.ps1`.
- There is also prior art on the ScubaGoggles side: `Testing/Markdown/test_links.py` was called out in an earlier comment and still looks like a good reference for the extraction/validation flow.

My read is that the smallest path would be:
1. extend `ValidateMarkdowns.Tests.ps1` so it extracts links from each baseline markdown;
2. validate anchors/local relative links structurally first;
3. validate external `https` links with a retrying request helper and allow a small skip/allowlist for hosts that reject HEAD or rate-limit CI.

One thing I would explicitly include is the embedded links inside policy sections, not just the top-level resource lists. That matches the bug history described in the issue and keeps the test from missing the most user-visible links.

If you want, I can put together a first pass PR for the extraction and structural checks, then leave network-flake handling as a second pass.

**Comment by tkol2022** [comment]
@dagarwal-mitre This sounds like a test we could run at the end of the release, but not during every functional test because it would waste time and links don't change much. Maybe it would make sense to write a PowerShell script or Python that a separate GitHub action could call.

**Comment by dagarwal-ecs** [comment]
> [@dagarwal-mitre](https://github.com/dagarwal-mitre) This sounds like a test we could run at the end of the release, but not during every functional test because it would waste time and links don't change much. Maybe it would make sense to write a PowerShell script or Python that a separate GitHub action could call.

@tkol2022 Yup I agree, I would lean towards using Powershell for consistency since the rest of the codebase is all PS, even though I would prefer Python. I can make an issue to create a workflow for this, but one thing we will need to be careful of is hitting rate limits something we have experienced on the Google side. It might be good to do issue #494 as part of this issue to change links from in line to reference links.
